.github/workflows/ci.yml

```diff
@@ -0,0 +1,122 @@
+name: Build and run tests
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  ci_linux:
+    runs-on: ubuntu-22.04
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        variant:
+          - gcc
+          - clang
+          - gcc-arm32
+          - gcc-arm32-thumb
+          - gcc-aarch64
+          - msvc-wine
+
+        include:
+          - variant: gcc
+            deps: g++-12
+            cc: gcc-12
+            cxx: g++-12
+
+          - variant: clang
+            deps: clang-15
+            cc: clang-15
+            cxx: clang++-15
+
+          - variant: gcc-arm32
+            deps: g++-12-arm-linux-gnueabihf qemu-user-static
+            cc: arm-linux-gnueabihf-gcc-12
+            cxx: arm-linux-gnueabihf-g++-12
+            cxxflags: -static -marm -Wno-psabi
+            test_runner: qemu-arm-static
+
+          - variant: gcc-arm32-thumb
+            deps: g++-12-arm-linux-gnueabihf qemu-user-static
+            cc: arm-linux-gnueabihf-gcc-12
+            cxx: arm-linux-gnueabihf-g++-12
+            cxxflags: -static -mthumb -Wno-psabi
+            test_runner: qemu-arm-static
+
+          - variant: gcc-aarch64
+            deps: g++-12-aarch64-linux-gnu qemu-user-static
+            cc: aarch64-linux-gnu-gcc-12
+            cxx: aarch64-linux-gnu-g++-12
+            cxxflags: -static
+            test_runner: qemu-aarch64-static
+            
+          - variant: msvc-wine
+
+            # Note: windows workers are in short supply on github,
```

Comment:
They're not in that short of supply; I include Windows in the CI matrix for several of my python projects and haven't noticed more than a couple minutes' delay in practice.

Reply:
Mine have not been starting for half an hour, so I ran out of patience.